modules/react/text-input/lib/TextInput.tsx

```diff
@@ -21,74 +20,102 @@ export interface TextInputProps extends Themeable, GrowthBehavior {
+      display: 'none',
+    },
     '&::placeholder': {
-      color: inputColors.placeholder,
+      color: system.color.text.hint,
     },
     '&:hover, &.hover': {
-      borderColor: inputColors.hoverBorder,
+      borderColor: system.color.border.input.strong,
     },
-    '&:focus-visible:not([disabled]), &.focus:not([disabled]), &:focus:not([disabled])': {
-      borderColor: inputColors.focusBorder,
-      boxShadow: `inset 0 0 0 1px ${inputColors.focusBorder}`,
+    '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+      borderColor: brand.common.focusOutline,
+      boxShadow: `inset 0 0 0 1px ${cssVar(brand.common.focusOutline)}`,
       outline: 'none',
     },
-    '&:disabled': {
-      backgroundColor: inputColors.disabled.background,
-      borderColor: inputColors.disabled.border,
-      color: inputColors.disabled.text,
+    '&:disabled, .disabled': {
+      backgroundColor: system.color.bg.alt.softer,
+      borderColor: system.color.border.input.disabled,
+      color: system.color.text.disabled,
       '&::placeholder': {
-        color: inputColors.disabled.text,
+        color: system.color.text.disabled,
       },
     },
-    '::-ms-clear': {
-      display: 'none',
-    },
-  },
-  ({width}) => ({
-    minWidth: width || 280,
-    width,
   }),
-  ({grow}) =>
-    grow && {
-      width: '100%',
+  modifiers: {
+    grow: {
+      true: {
+        width: '100%',
+        resize: 'vertical',
+      },
     },
-  ({theme, error}) => {
-    return {
-      '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
-        borderColor: theme.canvas.palette.common.focusOutline,
-        boxShadow: `inset 0 0 0 1px ${theme.canvas.palette.common.focusOutline}`,
-        outline: 'none',
+
+    error: {
+      error: {
+        borderColor: brand.error.base,
+        boxShadow: `inset 0 0 0 ${px2rem(1)} ${brand.error.base}`,
+        '&:hover, &.hover': {
+          borderColor: brand.error.base,
+        },
+        '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+          borderColor: brand.error.base,
+          boxShadow: `inset 0 0 0 ${px2rem(1)} ${brand.error.base},
+        0 0 0 2px ${system.color.border.inverse},
+        0 0 0 4px ${brand.common.focusOutline}`,
+        },
+        '&:disabled, .disabled': {
+          borderColor: brand.error.base,
+        },
+      },
+      alert: {
+        borderColor: brand.alert.darkest,
+        boxShadow: `inset 0 0 0 ${px2rem(2)} ${brand.alert.base}`,
+        '&:hover, &.hover': {
+          borderColor: brand.alert.darkest,
+        },
+        '&:focus-visible:not([disabled]), &.focus:not([disabled])': {
+          borderColor: brand.alert.darkest,
+          boxShadow: `inset 0 0 0 ${px2rem(2)} ${brand.alert.base},
+        0 0 0 2px ${system.color.border.inverse},
+        0 0 0 4px ${brand.common.focusOutline}`,
+        },
+        '&:disabled, .disabled': {
+          borderColor: brand.alert.darkest,
+        },
```

Comment:
Seems like this styles are the same that in error, just some colors are different. What do you think about the same approach we have in checkbox? error colors set in parent components and this codeblock is repeated only once, if error has any value.

Reply:
hmm, there's no parent in this case, but happy to write is a different way, I did clean up the selectors!